### PR TITLE
add Black Duck risk profile rendering

### PR DESCRIPTION
**Pull request by claymccoy**

### CODE REVIEW

File: src/handlers/event/push/PushLifecycle.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ import {
 import {
     ApplicationNodeRenderer,
     BuildNodeRenderer,
-    CommitNodeRenderer,
+    CommitNodeRenderer, FingerprintNodeRenderer,
```
Comment (cdupuis) [inline src/handlers/event/push/PushLifecycle.ts]: that import should be on a new line

File: src/handlers/event/push/rendering/PushNodeRenderers.ts [hunk 2]
```diff
@@ -664,3 +664,35 @@ export class PullRequestNodeRenderer extends AbstractIdentifiableContribution
             });
     }
 }
+
+export class FingerprintNodeRenderer extends AbstractIdentifiableContribution
```
Comment (cdupuis) [inline src/handlers/event/push/rendering/PushNodeRenderers.ts]: isn't that really only dealing with BlackDuck? In that case `BlackDuckFingerprintNodeRenderer` might be a better name? WDYT?

File: src/handlers/event/push/rendering/PushNodeRenderers.ts [hunk 3]
```diff
@@ -664,3 +664,35 @@ export class PullRequestNodeRenderer extends AbstractIdentifiableContribution
...
+    implements NodeRenderer<graphql.PushToPushLifecycle.Push> {
+
+    constructor() {
+        super("fingerprint");
```
Comment (cdupuis) [inline src/handlers/event/push/rendering/PushNodeRenderers.ts]: see above

File: src/handlers/event/push/rendering/PushNodeRenderers.ts [hunk 4]
```diff
@@ -664,3 +664,35 @@ export class PullRequestNodeRenderer extends AbstractIdentifiableContribution
...
+    }
+
+    public render(push: graphql.PushToPushLifecycle.Push, actions: Action[], msg: SlackMessage,
+                  context: RendererContext): Promise<SlackMessage> {
+        const riskProfileFingerprint = push.after.fingerprints.find(s => s.name === "BlackDuckRiskProfile");
+        if (riskProfileFingerprint) {
+            const riskProfile = JSON.parse(riskProfileFingerprint.data);
+            const v = riskProfile.categories.VULNERABILITY;
+            const rpMsg = `Security Risks - ${v.HIGH} High, ${v.MEDIUM} Medium, ${v.LOW} Low`;
+            const attachment: Attachment = {
```
Comment (cdupuis) [inline src/handlers/event/push/rendering/PushNodeRenderers.ts]: Could we add a link to the report in BlackDuck?

Comment (claymccoy) [reply src/handlers/event/push/rendering/PushNodeRenderers.ts]: yes

Comment (claymccoy) [reply src/handlers/event/push/rendering/PushNodeRenderers.ts]: I have to parse some stuff out of the API and then reconstruct that. Let's get this in, and then I'll make that happen in the next phase.

**Pull request merged**